controllers/horizon_controller.go

```diff
@@ -502,37 +502,26 @@ func (r *HorizonReconciler) generateServiceConfigMaps(
 		return err
 	}
 
-	if instance.Spec.SharedMemcached == "" {
-		memcachedSvc, err = r.getMemcachedSvc(ctx, fmt.Sprintf("%s-memcached", instance.Name), h, instance.Namespace)
-		if err != nil {
-			instance.Status.Conditions.Set(condition.FalseCondition(
-				horizonv1alpha1.HorizonMemcachedReadyCondition,
-				condition.ErrorReason,
-				condition.SeverityError,
-				horizonv1alpha1.HorizonMemcachedServiceError,
-				err.Error()))
-			return err
-		}
-	} else {
-		memcachedSvc, err = r.getMemcachedSvc(ctx, instance.Spec.SharedMemcached, h, instance.Namespace)
-		if err != nil {
-			instance.Status.Conditions.Set(condition.FalseCondition(
-				horizonv1alpha1.HorizonMemcachedReadyCondition,
-				condition.ErrorReason,
-				condition.SeverityError,
-				horizonv1alpha1.HorizonMemcachedServiceError,
-				err.Error()))
-			return err
-		}
+	podList, err = getMemcachedPods(ctx, h, instance, fmt.Sprintf("%s-memcached", instance.Name))
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			horizonv1alpha1.HorizonMemcachedReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityError,
+			horizonv1alpha1.HorizonMemcachedServiceError,
+			err.Error()))
+		return err
 	}
 
+	memcachedAddress := formatMemcachedAddress(podList)
+
 	url := strings.TrimPrefix(instance.Status.HorizonEndpoints["public"], "http://")
 
 	templateParameters := map[string]interface{}{
 		"keystoneURL":        keystonePublicURL,
 		"horizonDebug":       instance.Spec.Debug,
 		"horizonEndpointUrl": url,
-		"memcachedSvc":       memcachedSvc,
+		"memcachedAddress":   memcachedAddress,
```

Comment:
This value contains not only address but also port. Probably memcachedServers would be better